application/src/ext/java/org/opentripplanner/ext/carpooling/util/BeelineEstimator.java

```diff
@@ -0,0 +1,109 @@
+package org.opentripplanner.ext.carpooling.util;
+
+import java.time.Duration;
+import java.util.List;
+import org.opentripplanner.framework.geometry.SphericalDistanceLibrary;
+import org.opentripplanner.framework.geometry.WgsCoordinate;
+
+/**
+ * Provides fast, low-resolution travel time estimates based on beeline (straight-line) distances.
+ * <p>
+ * Used as a heuristic to quickly reject incompatible insertion positions before
+ * performing expensive A* street routing. The estimates are intentionally optimistic
+ * (lower bounds) to ensure we never incorrectly reject valid insertions.
+ * <p>
+ * Formula: duration = (beeline_distance × detour_factor) / average_speed
+ * <p>
+ * The detour factor accounts for the fact that street routes are rarely straight lines.
+ * Typical values: 1.2-1.5, with 1.3 being a reasonable default for urban areas.
+ */
+public class BeelineEstimator {
+
+  /**
+   * Default detour factor: 1.3
+   * Assumes actual street routes are ~30% longer than straight-line distance.
+   * This is conservative - works well for most urban areas.
+   */
+  public static final double DEFAULT_DETOUR_FACTOR = 1.3;
+
+  /**
+   * Default average speed: 10 m/s (~36 km/h or ~22 mph)
+   * Typical urban carpooling speed accounting for traffic, turns, stops.
+   */
+  public static final double DEFAULT_SPEED_MPS = 10.0;
+
+  private final double detourFactor;
+  private final double speedMps;
```

Comment:
We have some naming convention that units should be separated with underscore. So `speed_mps`. Although, I don't know if we have ever included the units in speeds so maybe just remove the mps part and add it as javadoc.